packages/playwright-core/src/server/helper.ts

```diff
@@ -69,10 +70,11 @@ class Helper {
           reject(e);
         }
       }));
+    })).catch(error => {
+      dispose();
```

Comment:
Do it once in `Promise.finally` and remove `removeEventListeners` calls from lines 66 and 69 above.